### PR TITLE
Clean up CSS background color styles

### PR DESCRIPTION
**Pull request by sqs**
Standardizes how components declare background colors. No components declare their own custom theme colors for backgrounds anymore; now they use CSS custom properties that can change based on the user's selected theme.

There will be minor changes to the background colors used in various components, but no (intentional) major changes. If we want to tweak colors further, we'll just need to introduce more variables that themes define (which is super easy).

Similar to #14847 and #14829.

### CODE REVIEW

**Comment by felixfbecker** [comment]
@sourcegraph/design I'd like to get some design eyes on this, could one of you check locally if the color changes are okay? I'll try to point out where colors changed.

**Comment by felixfbecker** [comment]
**⛔️ issue**: Selected folder text color in file tree is low-contrast in light theme (was white before)

image

**Comment by felixfbecker** [comment]
This seems to fix #14251 too

File: client/web/src/tree/Tree.scss [hunk 1]
```diff
@@ -18,28 +18,30 @@
...
 
         &--active {
-            color: #ffffff;
             width: 100%;
             display: table;
-            background-color: #1c7cd6 !important;
+            background-color: var(--primary) !important;
+        }
+        &--active &-contents {
+            color: #ffffff !important;
```
Comment (felixfbecker) [inline client/web/src/tree/Tree.scss:44]:
```suggestion
            color: $white !important;
```

**Comment by AlicjaSuska** [comment]
@felixfbecker agree - contrast is too low. We should either use white text there or coma back to the lighter selection background color.

**Pull request merged**